File: Project 4a/Step1.pl
% This Prolog file contains predicates related to manipulating and sorting names.
% William Hutto
% Programming Languages



% Define the variable Name with your first and last name
name(williamhutto).

% Predicate to sort and return name's characters in alphabetical order
sort_name_chars_sorted(Name, SortedChars) :-
    name(Name),                      % Retrieve the name
    atom_chars(Name, Chars),         % Convert name to list of characters
    msort(Chars, SortedChars).        % Sort the characters


% Predicate to reverse the characters of a name
reverse_name(Name, ReversedName) :-
    name(Name),                      % Retrieve the name
    atom_chars(Name, Chars),         % Convert name to list of characters
    reverse(Chars, ReversedChars),   % Reverse the characters
    atom_chars(ReversedName, ReversedChars). % Convert reversed characters to atom


% Predicate to sort the characters of a name in reverse order
reverse_sort_name(Name, ReverseSortedName) :-
    name(Name),                            % Retrieve the name
    atom_chars(Name, Chars),               % Convert name to list of characters
    msort(Chars, SortedChars),              % Sort the characters
    reverse(SortedChars, ReverseSortedChars), % Reverse the sorted characters
    atom_chars(ReverseSortedName, ReverseSortedChars). % Convert reversed sorted characters to atom

File: Project 5a/Step1.scala
/* 
* Scala Exploration: Step 1
* Define a class called Person with a string attribute for the name
* Create an instance of Person
* Set that instance's name to your first and last name.
* Display your name from the object
* Sort and display your name’s characters in alphabetical order
* Reverse and display your name
* Sort and display your name in reverse alphabetical order
* Author: William Hutto
* Date: 3/13/2024
* Class: Programming Languages
 */

// Define a class called Person with a string attribute for the name
class Person(var name: String)


@main def main(): Unit = { 

	// Create an instance of Person using your first name as the instance name. 
	val person = new Person("William")

	// Set that instance's name to your first and last name. 
	person.name = "William Hutto"

	// Display your name from the object
	println(person.name)

	// Sort and display your name’s characters in alphabetical order
	println(person.name.toCharArray.sorted.mkString)

	// Reverse and display your name
	println(person.name.reverse)

	// Sort and display your name in reverse alphabetical order
	println(person.name.toCharArray.sorted.reverse.mkString)
}